scanpy/external/tl/_harmony_timeseries.py

```diff
@@ -0,0 +1,462 @@
+"""Harmony time series for data visualization with augmented affinity matrix at discrete timepoints
+"""
+
+from scanpy import logging as logg
+import pandas as pd
+from collections import OrderedDict
+from itertools import chain
+import numpy as np
+import os
+
+
+def harmony_timeseries(data: list, **kargs):
+    """
+    Harmony time series for data visualization with augmented affinity matrix at discrete timepoints [Nowotschin18i]_.
+
+    Harmony time series is a framework for data visualization, trajectory detection and interpretation
+    for scRNA-seq data measured at discrete timepoints. Harmony constructs an augmented
+    affinity matrix by augmenting the kNN graph affinity matrix with mutually nearest
+    neighbors between successive time points. This augmented affinity matrix forms the
+    basis for generated a force directed layout for visualization and also serves as input
+    for computing the diffusion operator which can be used for trajectory detection using
+    `Palantir <https://github.com/dpeerlab/Palantir>`__.
+
+    .. note::
+        More information and bug reports `here <https://github.com/dpeerlab/Harmony>`__.
+
+    Parameters
+    ----------
+    data
+        list of :class:`~anndata.AnnData`, or cell x gene `.csv` filenames of count matrices from two time
+        points. Please ensure that replicates of the same time point are consecutive in order
+
+    timepoints
+        defaults to None, a list of timepoints at which each cell was measured. Important for Harmony augmented affinity matrix
+
+    sample_names
+        defaults to None, a list sample names in the form of timepoint_rep# of length `data`
+
+    log_transform
+        defaults to False, property setter passed to harmony
+
+
+    Returns
+    -------
+    `harmony_adata` of :class:`~anndata.AnnData` of normalized data:
```

Comment:
What does `harmony_adata` mean here? is this some subclass or the “name” of the return value? In that case please remove it.